backend/api/publishing.py

```
@@ -639,6 +639,19 @@ def _save_scenario_to_db(
+    # Prevent saving incomplete data during PDF parsing
+    # If we're updating an existing scenario and data is incomplete, skip the save
+    if scenario_id is not None:
+        scenes_count = len(actual_ai_result.get('scenes', []))
+        personas_count = len(actual_ai_result.get('key_figures', []))
+        
+        # Check if scenario exists and is in "creating" status
+        existing_scenario = db.query(Scenario).filter_by(id=scenario_id).first()
+        if existing_scenario and existing_scenario.status == "creating" and scenes_count == 0 and personas_count == 0:
+            debug_log(f"[SAVE_SKIP] Skipping save for scenario {scenario_id} - incomplete data during parsing (0 scenes, 0 personas)")
+            # Return the existing scenario without saving incomplete data
+            return existing_scenario, None, [], [], title
```

Ruff (0.14.3)

Undefined name `title`

(F821)